src/components/molecules/math-operation/math-operation.component.tsx

```diff
@@ -60,41 +59,49 @@ const MathOperationComponent: React.FC<MathOperationProps> = ({ operationLabel,
         onOperationClick(inputs, setResult, setErrorMessage);
     };
 
-    return (
-        <div className="math-operation-container">
-            {/* Nawigacja */}
-            <div className="navigation">
-                <Link to="/math">
-                    <Button label="⬅" />
-                </Link>
-
-                {operation && <Label text={operationNames[operation]} />}
-            </div>
+return (
+    <Paper elevation={3} sx={{ p: 4, 
+    width: '60vw', 
+    display: 'flex', 
+    flexDirection: 'column', 
+    alignItems: 'center', 
+    backgroundColor: '#CFEDDB' }}>
+        {/* Nawigacja */}
+        <BaseButton component={Link} to="/math">
+            <ArrowBackIosIcon />
+        </BaseButton>
 
-            {/* Pola wejściowe */}
-            <div className="input-fields">
-                {inputs.map((input, index) => (
-                    <Input
-                        key={index}
+        {operation && 
+            <Typography variant="h4" sx={{ textAlign: 'center', width: '100%', pt: 3 }}>
+                {operationNames[operation]}
```

Comment:
trochę bym to zabezpieczył -> czy jak odwołasz się do operacji, która nie istnieje to czy przypadkiem nie dostaniesz soczystego błędu który wywali całą aplikację?

bez Error Boundaries chyba nawet zrobi się biały ekran